docs/monitoring_logging.md

```diff
@@ -0,0 +1,188 @@
+
+##### Enabling Azure Monitor on AKS using the CLI
+
+
+```sh
+az aks enable-addons -a monitoring -n MyExistingManagedCluster -g MyExistingManagedClusterRG
+```
+
+##### Example - Viewing Analyzer Logs
+
+Run the following KQL query in Azure Logs:
+
+```sql
+let startTimestamp = ago(1d);
+let ContainerIDs = KubePodInventory
+| where TimeGenerated > startTimestamp
+| where ClusterId =~ "[YOUR_CLUSTER_ID]"
+| distinct ContainerID;
+ContainerLog
+| where ContainerID in (ContainerIDs) and Image contains "analyzer" 
+| project LogEntrySource, LogEntry, TimeGenerated, Computer, Image, Name, ContainerID
+| order by TimeGenerated desc
+| limit 200
+```
+
+### Logging with Elasticsearch, Kibana, FluentD
+
+Logs in presidio are outputted to stdrr and stdout as a standard of logging in 12 factor/microservices applications.  
+to store logs for long term retention and exploration during failures and RCA, use [elasticsearch](https://github.com/elastic/elasticsearch) or other document databases that are optimized to act as a search engine (solr, splunk, etc). elasticsearch logs are easily queried and visualized using [kibana](https://github.com/elastic/kibana) or [grafana](https://github.com/grafana/grafana).  
+Shipping logs from a microservices platform such as kubernetes to the logs database is done using a logs processor\forwarder such as the CNCF project [FluentD](https://www.fluentd.org/).  
+
+##### Enabling EFK on AKS
+
+The following section describes deploying an EFK (Elastic, FluentD, Kibana) stack to a development AKS cluster.  
+**Note that:** the following scripts do **not** fit a production environment in terms of security and scale.
+
+- Install elasticsearch
+
+```sh
+helm install stable/elasticsearch --name=elasticsearch --namespace logging --set client.replicas=1 --set master.replicas=1  --set cluster.env.MINIMUM_MASTER_NODES=1 --set cluster.env.RECOVER_AFTER_MASTER_NODES=1 --set cluster.env.EXPECTED_MASTER_NODES=1  --set data.replicas=1 --set data.heapSize=300m  --set master.persistence.storageClass=managed-premium --set data.persistence.storageClass=managed-premium
+```
+
+- Install fluent-bit (a lightweight fluentD log-forwarder)
+
+```sh
+helm install stable/fluent-bit --name=fluent-bit --namespace=logging --set backend.type=es --set backend.es.host=elasticsearch-client
+```
+
+- Install kibana
+
+```sh
+helm install stable/kibana --version 3.0.0  --name=kibana --namespace=logging  --set env.ELASTICSEARCH_URL=http://elasticsearch-client:9200 --set files."kibana\.yml"."elasticsearch\.hosts"=http://elasticsearch-client:9200 --set service.type=NodePort --set service.nodePort=31000
+```
+
+##### Example - Viewing Analyzer Logs
+
+- Open the kinana dashbaord
+
+```sh
+kubectl -n logging port-forward $(kubectl -n logging get pod -l app=kibana -o jsonpath='{.items[0].metadata.name}') 5601:5601
+```
+
+- Open your browser at http://localhost:5601
+
+- TODO: continue when we have the logs
```

Comment:
If this isn't finalize we should add a task instead of a TODO

Reply:
fixed